File: Leetcode/Array/238.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """

        answer = []
        tmp = 1
        for num in nums:
            answer.append(tmp)
            tmp *= num

        tmp = 1
        for i in reversed(range(len(nums))):
            answer[i] *= tmp
            tmp *= nums[i]

        return answer

File: Leetcode/Array/238.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function (nums) {
  let answer = [];
  let tmp = 1;

  for (let i = 0; i < nums.length; i++) {
    answer.push(tmp);
    tmp *= nums[i];
  }

  tmp = 1;
  for (let i = nums.length - 1; i >= 0; i--) {
    answer[i] *= tmp;
    tmp *= nums[i];
  }

  return answer;
};
